fpdf/table.py

```diff
@@ -0,0 +1,337 @@
+from dataclasses import dataclass
+from numbers import Number
+from typing import List, Union
+
+from .enums import Align, TableBordersLayout
+from .fonts import FontStyle
+
+
+DEFAULT_HEADINGS_STYLE = FontStyle(emphasis="BOLD")
+
+
+class Table:
+    """
+    Object that `fpdf.FPDF.table()` yields, used to build a table in the document.
+    Detailed usage documentation: https://pyfpdf.github.io/fpdf2/Tables.html
+    """
+
+    def __init__(
+        self,
+        fpdf,
+        rows=(),
+        *,
+        align="CENTER",
+        borders_layout=TableBordersLayout.ALL,
+        cell_fill_color=None,
```

Comment:
This is slightly confusing.

I would expect a "cell_fill_color" to apply to all cells.
There should probably be something like "alternate_cell_fill_color", whith the additional benefit that we then can explitly define both.

Took me a while to notice the difference between row and column coloring... :wink:
But since both are possible: Can we come up with a way to vary both at the same time? That would obviously require four colors, and maybe a lambda that returns one out of `range(4)`.

Or taking a step further back: If the goal is just to alternate colors, then we could skip the lambda altogether, and simply offer a pick of "ROWS", "COLUMNS", and "BOTH". Anything more complex than that is probably easier to imlement by setting each cell background individually instead of using a lambda.

Reply:
> Or taking a step further back: If the goal is just to alternate colors, then we could skip the lambda altogether, and simply offer a pick of "ROWS", "COLUMNS", and "BOTH". Anything more complex than that is probably easier to imlement by setting each cell background individually instead of using a lambda.

Yes, I agree with. Making the switch now.